src/mscorlib/shared/System/ReadOnlyMemory.cs

```diff
@@ -3,26 +3,29 @@
 // See the LICENSE file in the project root for more information.
 
 using System.Buffers;
-using EditorBrowsableAttribute = System.ComponentModel.EditorBrowsableAttribute;
-using EditorBrowsableState = System.ComponentModel.EditorBrowsableState;
 using System.Diagnostics;
-using System.Runtime;
 using System.Runtime.CompilerServices;
 using System.Runtime.InteropServices;
+using EditorBrowsableAttribute = System.ComponentModel.EditorBrowsableAttribute;
+using EditorBrowsableState = System.ComponentModel.EditorBrowsableState;
 
 namespace System
 {
+    /// <summary>
+    /// Represents a contiguous region of memory, similar to <see cref="ReadOnlySpan{T}"/>.
+    /// Unlike <see cref="ReadOnlySpan{T}"/>, it is not a byref-like type.
+    /// </summary>
     [DebuggerDisplay("{DebuggerDisplay,nq}")]
     [DebuggerTypeProxy(typeof(MemoryDebugView<>))]
     public readonly struct ReadOnlyMemory<T>
     {
         // NOTE: With the current implementation, Memory<T> and ReadOnlyMemory<T> must have the same layout,
         // as code uses Unsafe.As to cast between them.
 
-        // The highest order bit of _index is used to discern whether _arrayOrOwnedMemory is an array or an owned memory
-        // if (_index >> 31) == 1, object _arrayOrOwnedMemory is an OwnedMemory<T>
-        // else, object _arrayOrOwnedMemory is a T[]
-        private readonly object _arrayOrOwnedMemory;
+        // The highest order bit of _index is used to discern whether _object is an array/string or an owned memory
+        // if (_index >> 31) == 1, _object is an OwnedMemory<T>
+        // else, _object is a T[] or string
+        private readonly object _object;
         private readonly int _index;
         private readonly int _length;
```

Comment:
I just realized we could use the sign of the _length to tell if the _object is a string, i.e. we do have one more bit to spare. Would we want to do that?

Reply:
It's good to have the option, but it's not clear to me that's faster; a type check for string is quite fast, and we'll need to do the cast from object anyway if it is a string.  Let's start with the type check, and we can always change the implementation detail later.